### PR TITLE
Feature test block g

### PR DESCRIPTION
**Pull request by monte97**
Add first tests for BlockG
- G[V]
- distanceTo
- distanceBetween
- broadcast
- channel

### CODE REVIEW

File: tests/src/test/scala/it/unibo/scafi/test/functional/TestScafiTestUtils.scala [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    assert(net.neighbourhood(2) == Set(1,4,5))
+    assert(net.neighbourhood(3) == Set(0,1,4,7))
+    assert(net.neighbourhood(4) == Set(0,1,2,3,5,7,8))
+    assert(net.neighbourhood(5) == Set(1,2,4,7,8))
+    assert(net.neighbourhood(6).isEmpty)
+    assert(net.neighbourhood(7) == Set(3,4,5,8))
+    assert(net.neighbourhood(8) == Set(4,5,7))
+  }
+
+
```
Comment (metaphori) [inline tests/src/test/scala/it/unibo/scafi/test/functional/TestScafiTestUtils.scala:73]: Remove these empty lines

File: tests/src/test/scala/it/unibo/scafi/test/functional/ScafiTestUtils.scala [hunk 2]
```diff
@@ -89,4 +93,25 @@ object ScafiTestUtils {
...
+          detachedNodesCoords
+            .find(_ == (a,b))
+            .map{_ =>
+              lastDetachedPosition = (lastDetachedPosition._1 - rng, lastDetachedPosition._2 - rng)
+              lastDetachedPosition
+            }
+            .getOrElse[(Double, Double)](px, py)
+        ), rng = rng)
+  }
+}
```
Comment (metaphori) [inline tests/src/test/scala/it/unibo/scafi/test/functional/ScafiTestUtils.scala:117]: Add newline

File: tests/src/test/scala/it/unibo/scafi/test/functional/TestScafiTestUtils.scala [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    assert(net.neighbourhood(0).isEmpty)
+    assert(net.neighbourhood(1) == Set(3,4,5))
+    assert(net.neighbourhood(2).isEmpty)
+    assert(net.neighbourhood(3) == Set(1,4,7))
+    assert(net.neighbourhood(4) == Set(1,3,5,7))
+    assert(net.neighbourhood(5) == Set(1,4,7))
+    assert(net.neighbourhood(6).isEmpty)
+    assert(net.neighbourhood(7) == Set(3,4,5))
+    assert(net.neighbourhood(8).isEmpty)
```
Comment (metaphori) [inline tests/src/test/scala/it/unibo/scafi/test/functional/TestScafiTestUtils.scala:24]: I do not remember. There was a reason for which in this and the following testcases we do not use a single assertion "as usual" where the assertion value is structured using a "grid" shape through line breaks?

Comment (monte97) [reply tests/src/test/scala/it/unibo/scafi/test/functional/TestScafiTestUtils.scala:24]: In this case I'm testing a property of the network, the helper function `assertNetworkValues` checks against node's value from the export. 

Using 
```scala

    assertNetworkValues((0 to 8).zip(List(
      net.neighbourhood(0).isEmpty, net.neighbourhood(1) == Set(3,4,5), net.neighbourhood(2).isEmpty,
      net.neighbourhood(3) == Set(3,4,5),  net.neighbourhood(4) == Set(3,4,5),  net.neighbourhood(5) == Set(3,4,5),
      net.neighbourhood(6).isEmpty, net.neighbourhood(7) == Set(3,4,5), net.neighbourhood(8).isEmpty
    )).toMap)(net)
```

Will result in the following netwok:
```
Actual network: 
_	_	_	
_	_	_	
_	_	_
```

**Pull request merged**